receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet.go

```diff
@@ -0,0 +1,88 @@
+package k8swindows
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/kubeletutil"
+
+	"go.uber.org/zap"
+	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
+)
+
+type kubeletSummaryProvider struct {
+	logger   *zap.Logger
+	hostIP   string
+	hostPort string
+	client   *kubeletutil.KubeletClient
+	hostInfo host.Info
+}
+
+func new(logger *zap.Logger, info host.Info) (*kubeletSummaryProvider, error) {
+	hostIP := os.Getenv("HOST_IP")
+	kclient, err := kubeletutil.NewKubeletClient(hostIP, ci.KubeSecurePort, logger)
+	if err != nil {
+		return nil, errors.New("failed to initialize kubelet client")
```

Comment:
To improve error handling and make debugging easier, can we please include the original error details so that a user will know what happened and how to rectify it?

Reply:
Addressed